### PR TITLE
fix: `--ignore-pattern` in flat config mode should be relative to `cwd`

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Closes https://github.com/eslint/eslint/issues/16264

#### What changes did you make? (Give an overview)

After fixing https://github.com/eslint/eslint/issues/16300, `--ignore-pattern` became relative to `configs.basePath`, because they were mapped to `overrideConfig.ignores`. I updated `cli.js` to map them to the `ignorePatterns` constructor option instead, so now they will be again relative to `cwd`.

#### Is there anything you'd like reviewers to focus on?

The removed conditional in `flat-eslint.js`.

### CODE REVIEW

File: lib/eslint/flat-eslint.js [hunk 1]
```diff
@@ -362,17 +362,6 @@ async function calculateConfigArray(eslint, {
...
-             * Ignore patterns are considered relative to a directory
-             * when the pattern contains a slash in a position other
-             * than the last character. If that's the case, we need to
-             * add the relative ignore path to the current pattern to
-             * get the correct behavior. Otherwise, no change is needed.
-             */
-            if (!basePattern.includes("/") || basePattern.endsWith("/")) {
-                return pattern;
-            }
-
```
Comment (mdjermanovic) [inline lib/eslint/flat-eslint.js:375]: I'm not sure why we had this special case. It could be a leftover from the version where patterns had .gitignore semantics (e.g, `*.js` matches at any level).

File: lib/cli.js [hunk 2]
```diff
@@ -193,6 +187,10 @@ async function translateOptions({
         options.useEslintrc = eslintrc;
         options.extensions = ext;
         options.ignorePath = ignorePath;
+    } else {
+        if (ignorePattern) {
```
Comment (nzakas) [inline lib/cli.js:191]: Can we rearrange the conditionals in these blocks? It’s a bit hard to follow an else after a not equals. Maybe structure as “if configType is flat, then add ignorePatterns, else do the other stuff”?

Also, I don’t think you need to check for ignorePattern before adding ignorePatterns to options.

File: lib/eslint/flat-eslint.js (hunk 1 above)
Comment (nzakas) [reply lib/eslint/flat-eslint.js:375]: Yup, I’m pretty sure that’s left over from gitignore compatibility.

**Review by nzakas — CHANGES_REQUESTED** [review]

File: lib/cli.js (hunk 2 above)
Comment (mdjermanovic) [reply lib/cli.js]: Done, thanks for the suggestion!

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!
